src/boot/efi/boot.c

```diff
@@ -101,6 +101,53 @@ enum {
         TIMEOUT_TYPE_MAX    = UINT32_MAX,
 };
 
+#define _returns_nonnull_ __attribute__((__returns_nonnull__))
+#define _allocate_pool_ __attribute__((__malloc__, __malloc__(FreePool, 1), __warn_unused_result__))
+#define assert_alloc_return(p) ({ \
+        if (_likely_(p))          \
+                return p;         \
+        log_oom();                \
+        __builtin_trap();         \
+})
+
+_allocate_pool_ _returns_nonnull_ void *new_nofail(UINTN size) {
```

Comment:
in userspace new() is a macro that operates in a typesafe fashion, i.e. takes a type as first param, and then allocates one or more objects of this type. we should define xnew() in uefi the same way.